modules/publicize/class-async-publicize.php

```diff
@@ -0,0 +1,343 @@
+ *
+ * This class sets up REST endpoints for publishing posts and then
+ * later sharing them with Publicize.
+ *
+ * @since 5.9.1
+ */
+class Async_Publicize {
+	/**
+	 * Instance of publicize class, used to access helper methods.
+	 *
+	 * @since 5.9.1
+	 * @var  Publicize $publicize
+	 */
+	protected $publicize;
+
+	/**
+	 * Meta key name to flag publishing posts that should not be publicized.
+	 *
+	 * @since 5.9.1
+	 * @var string $PUBLICIZE_BLOCK_META_KEY
+	 */
+	const PUBLICIZE_BLOCK_META_KEY = '_no_publicize';
+
+	/**
+	 * Time out in seconds before PUBLICIZE_BLOCK_META_KEY meta key expires.
+	 *
+	 * @since 5.9.1
+	 * @var int META_KEY_TIMEOUT_SECONDS
+	 */
+	const META_KEY_TIMEOUT_SECONDS = 60 * 5;
+
+	/**
+	 * Time out in seconds before WPCOM REST calls timeout.
+	 *
+	 * @since 5.9.1
+	 * @var int WPCOM_REST_TIMEOUT
+	 */
+	const WPCOM_REST_TIMEOUT = 90;
+
+	/**
+	 * WPCOM REST API version string.
+	 *
+	 * @since 5.9.1
+	 * @var string API_VERSION
+	 */
+	const API_VERSION = 'v2';
+
+	/**
+	 * Base URL to WPCOM REST API.
+	 *
+	 * @since 5.9.1
+	 * @var string API_BASE_URL
+	 */
+	const API_BASE_URL = 'https://public-api.wordpress.com/wpcom/';
+
+	/**
+	 * Cosntructor for Async_Publicize
+	 *
+	 * Set up hooks to extend legacy Publicize behavior.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @param Publicize $publicize Instance of main module class, used to access helper methods.
+	 */
+	public function __construct( $publicize ) {
+		$this->publicize = $publicize;
+
+		add_action( 'rest_api_init', function () {
+			register_rest_route( 'publicize/', '/posts/(?P<post_id>\d+)/flag-no-publicize', array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'flag_post_no_publicize' ),
+				'post_id'             => array(
+					'validate_post_id' => function( $param, $request, $key ) {
+						return is_int( $param );
+					},
+				),
+				'permission_callback' => function () {
+					return current_user_can( 'publish_posts' );
+				},
+			) );
+		} );
+
+		add_action( 'rest_api_init', function () {
+			register_rest_route( 'publicize/', '/posts/(?P<post_id>\d+)/publicize', array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'publicize_post' ),
+				'post_id'             => array(
+					'validate_post_id' => function( $param, $request, $key ) {
+						return is_int( $param );
+					},
+				),
+				'permission_callback' => function () {
+					return current_user_can( 'publish_posts' );
+				},
+			) );
+		} );
+
+		// This filter is documented in publicize-jetpack.php.
+		add_filter( 'publicize_should_publicize_published_post', array( $this, 'can_publicize' ), 1, 2 );
+
+		// Setup callback to do cleanup after the post has actually been published.
+		add_action( 'jetpack_published_post', array( $this, 'post_publish_cleanup' ), 10, 2 );
+
+		// Do post edit page specific setup.
+		add_action( 'admin_enqueue_scripts', array( $this, 'post_page_enqueue' ) );
```

Comment:
I believe you would need to use the `enqueue_block_editor_assets` hook here, so your script only gets enqueued in the Gutenberg editor:
https://github.com/WordPress/gutenberg/tree/v2.4.0/blocks#getting-started

Reply:
Nice! I wasn't aware of `enqueue_block_editor_assets`